### ISSUE TITLE
getRatings() - all plots

### ISSUE THREAD

**Issue by diecode** [IntellectualCrafters/PlotSquared#337]
I would like make a "favorites plots", but only there is getRatings(Plot plot). I think this is not too efficiency with lots of plot. ( Lots of queries )

Can you make getRatings() which return all plots' ratings? Maybe ```HashMap<Plot, Double>```

**Comment by boy0001** [comment]
As of 8e499dffb28b5d76d7db2c6db970e5fc2a0a74a5, ratings are now cached in: `plot.settings.ratings`. You can disable this caching in the settings.yml.

It will be null if there are no ratings for the plot, it will also may be subject to change in the future.

**Comment by boy0001** [comment]
Oh, here's a snippet of code taken from the `/plot list top` command which sorts plots by popularity:

``` java
plots = new ArrayList<>(PS.get().getPlots());
Collections.sort(plots, new Comparator<Plot>() {
    @Override
    public int compare(Plot p1, Plot p2) {
        double v1 = 0;
        double v2 = 0;
        if (p1.settings.ratings != null && p1.settings.ratings.size() > 0) {
            for (Entry<UUID, Integer> entry : p1.settings.ratings.entrySet()) {
                v1 += entry.getValue() * entry.getValue();
            }
            v1 /= p1.settings.ratings.size();
            v2 += p2.settings.ratings.size();
        }
        if (p2.settings.ratings != null && p2.settings.ratings.size() > 0) {
            for (Entry<UUID, Integer> entry : p2.settings.ratings.entrySet()) {
                v2 += entry.getValue() * entry.getValue();
            }
            v2 /= p2.settings.ratings.size();
            v2 += p2.settings.ratings.size();
        }
        if (v2 == v1 && v2 != 0) {
            return p2.settings.ratings.size() - p1.settings.ratings.size();
        }
        return (int) Math.signum(v2 - v1);
    }
});
```

**Comment by diecode** [comment]
Thanks the solution. I get this when I use "/plots list top" command: http://gyazo.com/769acf2fb09649b10087d2725d50103f

**Comment by boy0001** [comment]
It was an issue with the json chat using imports that were named differently in 1.7.10-. This has been fixed as of 7f5eaa83fd3a12ab9de19c77e92199823bc58904 in which a `plot.getRatings()` method was also added which will support rating categories when they are added.
